pulpcore/cli/core/pulp_import.py

```diff
@@ -0,0 +1,83 @@
+import gettext
+from typing import Any, Dict
+
+import click
+
+from pulpcore.cli.common.context import (
+    DEFAULT_LIMIT,
+    PulpContext,
+    pass_entity_context,
+    pass_pulp_context,
+)
+from pulpcore.cli.common.generic import Mutex, destroy_command, href_option, show_command
+from pulpcore.cli.core.context import PulpImportContext, PulpImporterContext
+
+_ = gettext.gettext
+
+
+@click.group(name="import")
+def pulp_import() -> None:
+    pass
+
+
+@pulp_import.group()
+@pass_pulp_context
+@click.pass_context
+def pulp(ctx: click.Context, pulp_ctx: PulpContext) -> None:
+    ctx.obj = PulpImportContext(pulp_ctx)
+
+
+lookup_options = [href_option]
+
+pulp.add_command(show_command(decorators=lookup_options))
+pulp.add_command(destroy_command(decorators=lookup_options))
+
+
+@pulp.command()
+@click.option("--importer", type=str, required=True, help=_("Name of owning PulpImporter"))
+@click.option(
+    "--limit", default=DEFAULT_LIMIT, type=int, help=_("Limit the number of imports to show.")
+)
+@click.option("--offset", default=0, type=int, help=_("Skip a number of imports to show."))
+@pass_entity_context
+@pass_pulp_context
+def list(
+    pulp_ctx: PulpContext,
+    import_ctx: PulpImportContext,
+    importer: str,
+    limit: int,
+    offset: int,
+    **kwargs: Any
+) -> None:
+    params = {k: v for k, v in kwargs.items() if v is not None}
+    importer_ctx = PulpImporterContext(pulp_ctx)
+    import_ctx.importer = importer_ctx.find(name=importer)
+    result = import_ctx.list(limit=limit, offset=offset, parameters=params)
+    pulp_ctx.output_result(result)
```

Comment:
This can be replaced with the list_command factory if you create a callback for the `importer` option and pass that option in as a decorator.  Check out the [repository callback](https://github.com/pulp/pulp-cli/blob/develop/pulpcore/cli/common/generic.py#L81)

Reply:
#101 asks to refactor the import/export commands. This PR is to add import-functionality that works the way export currently does. We can address all the comments here - but as part of #101.

Right now, I a) need access to import, b) quickly, and c) in a way that's consistent with export. All of the comments here, apply to the export-side code that this mirrors, we should think about a better approach for and then fix both sides at the same time.

Reply:
(my comment above was in response to https://github.com/pulp/pulp-cli/pull/195#pullrequestreview-630491374 , apologies.) Working on changing to use the factory as described here.

Reply:
@ggainey I started some work on refactoring export.py and in it I figured out the callback needed for using the list factory, should be applicable for import as well: https://github.com/gerrod3/pulp-cli/commit/73dc0998e52941c8a1fb5363dc5c620713468597